cmd/dep/init.go

```diff
@@ -68,6 +81,12 @@ func (cmd *initCommand) Run(ctx *dep.Ctx, loggers *Loggers, args []string) error
 		root = ctx.WorkingDir
 	} else {
 		root = args[0]
+		if !filepath.IsAbs(args[0]) {
```

Comment:
I _think_ what we really want here is `filepath.Clean()`. That should do all the work you're manually doing here, plus a little more.